flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1257,6 +1257,40 @@ struct ValueOpCommon {
     llvm_unreachable("must be a constant op");
     return {};
   }
+
+  // Translate the arguments pertaining to any multidimensional array to row-major order for LLVM-IR.
+  static void toRowMajor(llvm::SmallVectorImpl<mlir::Attribute>& attrs,
+                         mlir::Type ty) {
+    const auto end = attrs.size();
+    for (std::remove_const_t<decltype(end)> i = 0; i < end; ++i) {
+      if (auto seq = ty.dyn_cast<fir::SequenceType>()) {
+        const auto dim = seq.getDimension();
+        if (dim > 1) {
+          std::reverse(attrs.begin() + i, attrs.begin() + i + dim);
+          i += dim - 1;
+        }
+        ty = seq.getEleTy();
+        continue;
```

Comment:
Why `if () {continue;}` instead of `if-else` chain?

Reply:
I vote for an `if-else` chain here on the grounds that it is easier to read and understand, in part because it uses fewer lines of code.  `continue` and its siblings are often very useful, but I don't think that is the case here.

Reply:
Agree.